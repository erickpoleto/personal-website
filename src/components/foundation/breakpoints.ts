import {DefaultTheme} from "styled-components";

export const breakpoints: DefaultTheme = {
    sm: "only screen and (max-width: 576px)",
    md: "only screen and (max-width: 768px)",
    ld: "only screen and (min-width: 992px)",
    xl: "only screen and (min-width: 1200px)",
    xxl: "only screen and (min-width: 1400px)",
}