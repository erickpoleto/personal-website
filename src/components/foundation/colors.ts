import {DefaultTheme} from "styled-components";

export const colors: DefaultTheme = {
    primary: '#FF5733',
    darkBlue: '#10164D',
    darkestBlue: '#0D1028',
    black600: "#313030",
    black800: "#1D1D1D",
    brown400: '#321811',
    brown600: '#2C1503',
    brown800: '#231709',
    grey200: '#e4e4e4',
}