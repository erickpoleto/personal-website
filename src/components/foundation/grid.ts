import {DefaultTheme} from "styled-components";

export const grid: DefaultTheme = {
    default: {
        gridTemplateColumns: "1fr repeat(12, minmax(auto, 4.2rem)) 1fr",
        gap: "1rem 1rem"
    },
    sm: {
        gridTemplateColumns: "2rem repeat(6, 1fr) 2rem",
        gap: "1rem 1rem"
    },
    md: {
        gridTemplateColumns: "1rem repeat(6, 1fr) 1rem",
        gap: "1rem 1rem"
    }
}