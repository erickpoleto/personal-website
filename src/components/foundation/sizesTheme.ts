import {DefaultTheme} from "styled-components";

export const sizesTheme: DefaultTheme = {
    fullscreen: {
        height: '100vh'
    }
}