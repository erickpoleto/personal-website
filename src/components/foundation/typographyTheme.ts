import {DefaultTheme} from "styled-components";

export const typography: DefaultTheme = {
    variant: {
        size: {
            title: "42px",
            headline1: "24px",
            headline2: "27px",
            headline3: "34px",
            body1: "13px",
            body2: "14.3px",
            body3: "15.5px"
        },
        mediaSize: {
            title: "30px",
            headline1: "22px",
            headline2: "25.3px",
            body1: "13px",
            body2: "15px",
            body3: "17px"
        }
    },
    bold: {
        light: "300",
        medium: "500",
        hard: "bold"
    }
}