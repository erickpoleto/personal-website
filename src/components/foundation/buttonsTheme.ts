import {DefaultTheme} from "styled-components";
import {typography} from "./typographyTheme";

export const buttonsTheme: DefaultTheme = {
    sizes: {
        sm: '7px 17px',
        md: '8px 22px',
        ld: '9px 27px'
    }
}