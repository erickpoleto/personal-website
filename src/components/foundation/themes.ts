import {DefaultTheme} from "styled-components";
import {typography} from "./typography/typographyTheme";
import {colors} from "./colors/colors";
import {buttonColors} from "./colors/buttonColors";

export const theme: DefaultTheme = {
    colors: colors,
    buttonColors: buttonColors,
    typography: typography
}