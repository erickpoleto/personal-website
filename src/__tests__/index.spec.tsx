import React, {ReactElement} from 'react';
import renderer from 'react-test-renderer'
import Home from "../pages";
import 'jest-styled-components'
import {theme} from "../components/foundation/themes";
import {ThemeProvider} from "styled-components";

const renderWithThemeProvider = (container: ReactElement) => {
    return <ThemeProvider theme={theme}>{container}</ThemeProvider>
}

test("render 'Typography' element", () => {
    const wrapped = renderWithThemeProvider(<Home/>);
    const tree = renderer.create(wrapped).toJSON();
    expect(tree.children).toMatchSnapshot();
});