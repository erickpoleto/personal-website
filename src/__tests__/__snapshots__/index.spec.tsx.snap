// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 'Typography' element 1`] = `
Array [
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  border: none;
  box-shadow: 1px 1px 3px black;
  padding: 5px 20px;
  background: linear-gradient(0.25turn,#FF5733,#FD9883);
}

.c3:active {
  opacity: 0.8;
  -webkit-animation: fade 200ms ease-out forwards;
  animation: fade 200ms ease-out forwards;
}

.c2 {
  font-size: 42px;
  font-weight: bold;
  color: white;
}

<div
    className="c0"
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="c1"
    >
      <text
        className="c2"
        color="white"
      >
        Welcome
      </text>
      <button
        className="c3"
      >
        <text
          className="c2"
          color="white"
        >
          more about
        </text>
      </button>
    </div>
  </div>,
]
`;
